File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        for (int n : nums) {
            ++freq[n];
        }
        vector<pair<int, int>> swapped;
        for (const auto& [key, value] : freq) {
            swapped.push_back({value, key});
        }
        std::ranges::sort(
            swapped, [](const pair<int, int>& a, const pair<int, int>& b) {
                return a.first > b.first;
            }
        );
        vector<int> out(k);
        for (int i = 0; i < k; ++i) {
            out[i] = swapped[i].second;
        }
        return out;
    }

};

File: Data Structures & Algorithms/three-integer-sum/submission-7.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        result, n = [], len(nums)
        for k in range(n-2):
            if nums[k] > 0:
                break
            if k > 0 and nums[k] == nums[k - 1]:
                continue

            l, r = k + 1 , n-1
            while l < r:
                total = nums[k] + nums[l] + nums[r]
                if total < 0:
                    l += 1
                elif total > 0:
                    r -= 1
                else:
                    result.append([nums[k], nums[l], nums[r]])
                    l += 1
                    r -= 1

                    while l < r and nums[l] == nums[l - 1]:
                        l += 1
                    while l < r and nums[r] == nums[r + 1]:
                        r -= 1

        return result